### ISSUE TITLE
Tailwind not rendering, and no real time refresh

### ISSUE THREAD

**Issue by thedivtagguy** [blackbaba/Flask-Tailwind-Starter#1]
Hi! Thank you for providing a starter! I'm just getting started with Flask but am familiar with Tailwind, so this seems to be a good starting point. I set it up like you specified but when I'm trying to add the following code, I am getting unformatted html output. 

`index.html` :
```html
{% extends "base.html" %}

{% block content %}
<div class="bg-pink-500 text-black px-6 py-2 shadow">
<h1>Hello World</h1>
    
</div>

<div class="flex flex-col w-full max-w-md px-4 py-8 bg-white rounded-lg shadow dark:bg-gray-800 sm:px-6 md:px-8 lg:px-10">
    <div class="self-center mb-6 text-xl font-light text-gray-600 sm:text-2xl dark:text-white">
        Login To Your Account
    </div>
    <div class="mt-8">
        <form action="#" autoComplete="off">
            <div class="flex flex-col mb-2">
                <div class="flex relative ">
                    <span class="rounded-l-md inline-flex  items-center px-3 border-t bg-white border-l border-b  border-gray-300 text-gray-500 shadow-sm text-sm">
                        <svg width="15" height="15" fill="currentColor" viewBox="0 0 1792 1792" xmlns="http://www.w3.org/2000/svg">
                            <path d="M1792 710v794q0 66-47 113t-113 47h-1472q-66 0-113-47t-47-113v-794q44 49 101 87 362 246 497 345 57 42 92.5 65.5t94.5 48 110 24.5h2q51 0 110-24.5t94.5-48 92.5-65.5q170-123 498-345 57-39 100-87zm0-294q0 79-49 151t-122 123q-376 261-468 325-10 7-42.5 30.5t-54 38-52 32.5-57.5 27-50 9h-2q-23 0-50-9t-57.5-27-52-32.5-54-38-42.5-30.5q-91-64-262-182.5t-205-142.5q-62-42-117-115.5t-55-136.5q0-78 41.5-130t118.5-52h1472q65 0 112.5 47t47.5 113z">
                            </path>
                        </svg>
                    </span>
                    <input type="text" id="sign-in-email" class=" rounded-r-lg flex-1 appearance-none border border-gray-300 w-full py-2 px-4 bg-white text-gray-700 placeholder-gray-400 shadow-sm text-base focus:outline-none focus:ring-2 focus:ring-purple-600 focus:border-transparent" placeholder="Your email"/>
                    </div>
                </div>
                <div class="flex flex-col mb-6">
                    <div class="flex relative ">
                        <span class="rounded-l-md inline-flex  items-center px-3 border-t bg-white border-l border-b  border-gray-300 text-gray-500 shadow-sm text-sm">
                            <svg width="15" height="15" fill="currentColor" viewBox="0 0 1792 1792" xmlns="http://www.w3.org/2000/svg">
                                <path d="M1376 768q40 0 68 28t28 68v576q0 40-28 68t-68 28h-960q-40 0-68-28t-28-68v-576q0-40 28-68t68-28h32v-320q0-185 131.5-316.5t316.5-131.5 316.5 131.5 131.5 316.5q0 26-19 45t-45 19h-64q-26 0-45-19t-19-45q0-106-75-181t-181-75-181 75-75 181v320h736z">
                                </path>
                            </svg>
                        </span>
                        <input type="password" id="sign-in-email" class=" rounded-r-lg flex-1 appearance-none border border-gray-300 w-full py-2 px-4 bg-white text-gray-700 placeholder-gray-400 shadow-sm text-base focus:outline-none focus:ring-2 focus:ring-purple-600 focus:border-transparent" placeholder="Your password"/>
                        </div>
                    </div>
                    <div class="flex items-center mb-6 -mt-4">
                        <div class="flex ml-auto">
                            <a href="#" class="inline-flex text-xs font-thin text-gray-500 sm:text-sm dark:text-gray-100 hover:text-gray-700 dark:hover:text-white">
                                Forgot Your Password?
                            </a>
                        </div>
                    </div>
                    <div class="flex w-full">
                        <button type="submit" class="py-2 px-4  bg-purple-600 hover:bg-purple-700 focus:ring-purple-500 focus:ring-offset-purple-200 text-white w-full transition ease-in duration-200 text-center text-base font-semibold shadow-md focus:outline-none focus:ring-2 focus:ring-offset-2  rounded-lg ">
                            Login
                        </button>
                    </div>
                </form>
            </div>
            <div class="flex items-center justify-center mt-6">
                <a href="#" target="_blank" class="inline-flex items-center text-xs font-thin text-center text-gray-500 hover:text-gray-700 dark:text-gray-100 dark:hover:text-white">
                    <span class="ml-2">
                        You don&#x27;t have an account?
                    </span>
                </a>
            </div>
        </div>


{% endblock content %}

{% block scripts %}
{{ super() }}


{% endblock scripts %}
```

Output: 

![image](https://user-images.githubusercontent.com/6006888/121781456-714e8f80-cbc2-11eb-9da8-b8c121cd0b93.png)

Additionally, when I try to run `npm run dev`, I get the following error: 
```
(env) PS D:\Gatsby\form> npm run dev

> flask-tailwind-starter@1.0.0 dev D:\Gatsby\form
> TAILWIND_MODE=watch NODE_ENV=development postcss ./app/static/src/css/tailwind.css -o ./app/static/dist/css/main.css -w
'TAILWIND_MODE' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! flask-tailwind-starter@1.0.0 dev: `TAILWIND_MODE=watch NODE_ENV=development postcss ./app/static/src/css/tailwind.css -o ./app/static/dist/css/main.css -w`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the flask-tailwind-starter@1.0.0 dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\LENOVO\AppData\Roaming\npm-cache\_logs\2021-06-12T15_33_12_455Z-debug.log
```

**Comment by blackbaba** [comment]
@thedivtagguy Let me try and run it now and see whats going on. Ill respond back here

**Comment by blackbaba** [comment]
@thedivtagguy Okay. I have just pushed a fix. I did my development on a mac and it seems you are on windows (?).
You just have to `git pull` and then do `npm install` again and everything should work fine.

**Cause**
The `npm run dev` command and other script commands in `package.json` are settings 2 ENV variables called `TAILWIND MODE` and `NODE_ENV`.

These 2 variables as they are set in the command require an additional package on windows (cross-env).
I have added the package to `package.json` file so its all good now.

**Real Time Refresh Issue**
Tailwind generates your styles in real time and updates the CSS file. 
The Flask Debug mode is ON so since Flask is serving your HTML/CSS files it will pick up the changes. 
But you still have to refresh your browser (**do CTRL+SHIFT+R to clear cache on Chrome**) to see the updated CSS served by Flask with each change since Flask doesnt do hot reload (unlike the hot reload  in Vue/React).

---
I just did  a fresh git clone on Windows and everything seems to be working now.
Let me know if it works for you.

Cheers
Amrinder
![Capture](https://user-images.githubusercontent.com/22995397/121801722-5ab04300-cc6b-11eb-992c-73c5e13e2cf2.JPG)
